geoh5py/objects/geo_image.py

```diff
@@ -348,37 +307,34 @@ def georeferencing_from_tiff(self):
         Get the geographic information from the PIL Image to georeference it.
         Run the georeference() method of the object.
         """
-        if self.tag is None:
+        if self.tag is None or self.image is None:
             raise AttributeError("The image is not georeferenced")
 
         try:
-            # get geographic information
-            u_origin = float(self.tag[33922][3])
-            v_origin = float(self.tag[33922][4])
-            u_cell_size = float(self.tag[33550][0])
-            v_cell_size = float(self.tag[33550][1])
-            u_count = float(self.tag[256][0])
-            v_count = float(self.tag[257][0])
-
-            # todo: not anymore because it's not always a rectangle
-            u_oposite = float(u_origin + u_cell_size * u_count)
-            v_oposite = float(v_origin - v_cell_size * v_count)
+            # get the tag values
+            tie_points = self._parse_tie_points(self.tag[33922])
+            u_count = int(self.tag[256][0])
+            v_count = int(self.tag[257][0])
+            u_cell_size = self.tag[33550][0]
+            v_cell_size = self.tag[33550][1]
 
             # prepare georeferencing
-            reference = np.array([[0.0, v_count], [u_count, v_count], [u_count, 0.0]])
+            if u_count != self.u_count or v_count != self.v_count:
+                raise ValueError("The image size does not match the size in the tag.")
 
-            locations = np.array(
-                [
-                    [u_origin, v_origin, 0.0],
-                    [u_oposite, v_origin, 0.0],
-                    [u_oposite, v_oposite, 0.0],
-                ]
+            self.vertices = self._compute_image_corners(
+                tie_points, u_cell_size, v_cell_size
             )
 
-            # georeference the raster
-            self.georeference(reference, locations)
-        except KeyError:
-            warnings.warn("The 'tif.' image has no referencing information.")
+        except KeyError as error:
```

Comment:
Try to be more precise with the try/except.  You run the risk of catching a stray key/value error in all that code that then gives a warning that makes no sense.